tests/jballerina-unit-test/src/test/resources/test-src/error/distinct_error_test.bal

```diff
@@ -8,3 +8,31 @@ function testFooError() returns Foo {
 
     return foo;
 }
+
+public type GraphAPIError distinct error<GraphAPIErrorDetails>;
+public type GraphAPIErrorDetails record {|
+    string code;
+    map<anydata> details;
+|};
+
+public function testFunctionCallInDetailArgExpr() {
+    json codeJson = "1234";
+    map<anydata> details = {};
+    var x = GraphAPIError("Concurrent graph modification", code = codeJson.toString(), details = details);
+    assertEquality(x.toString(), "error Concurrent graph modification code=1234 details=");
```

Comment:
Instead of using the `toString()` representation, shall we assert the message, code, and details maps separately?

Reply:
+1